toolbox/strings.py

```diff
@@ -102,6 +177,74 @@ def is_invite(string: str, *, fullmatch: bool = True) -> bool:
     return False
 
 
+def remove_markdown(content: str, formats: MarkdownFormat = MarkdownFormat.ALL) -> str:
+    """
+    Removes the markdown formatting from Discord messages.
+
+    Parameters
+    ----------
+    content : str
+        The `str` object, which needs their content cleaned from Discord's markdown formatting.
+    formats : MarkdownFormat
+        The `IntFlag` of the formatting that needs to be removed. Default is `MarkdownFormat.ALL`
+        Multiple can be supplied by using bitwise OR.
+
+    Returns
+    -------
+    str
+        The cleaned string without markdown formatting.
+    """
+    code_block_matches = []
+    for format, (regex, replace) in FORMAT_DICT.items():
+        if formats & format:
+            if format & MarkdownFormat.MULTI_CODE_BLOCK or format & MarkdownFormat.CODE_BLOCK:
+                code_block_matches += re.findall(regex, content)
+            matches = re.findall(regex, content)
+            if not code_block_matches:
+                for match in matches:
+                    if format & MarkdownFormat.MULTI_QUOTE or format & MarkdownFormat.QUOTE:
+                        content = remove_quote(content, format)
+                        continue
+                    content = content.replace(match, match[replace:-replace], 1)
+            else:
+                for match in matches:
+                    if format & MarkdownFormat.MULTI_CODE_BLOCK or format & MarkdownFormat.CODE_BLOCK:
+                        content = content.replace(match, match[replace:-replace], 1)
+                        continue
+                    else:
+                        ignore = False
+                        for code_block_match in code_block_matches:
+                            if match in code_block_match:
+                                ignore = True
+                        if not ignore:
+                            content = content.replace(match, match[replace:-replace], 1)
+
+    return content
+
+
+def remove_quote(content: str, formats: MarkdownFormat) -> str:
```

Comment:
Feel like this could be marked as internal, but that's just me.
```suggestion
def _remove_quote(content: str, formats: MarkdownFormat) -> str:
```